pkg/tsdb/prometheus/querydata/exemplar/labels.go

```diff
@@ -23,9 +28,14 @@ func NewLabelTracker() LabelTracker {
 // so that they can be used to build the label fields in the exemplar frame
 func (l *labelTracker) Add(labels map[string]string) {
 	for k := range labels {
-		if _, ok := l.labelSet[k]; !ok {
-			l.labelSet[k] = struct{}{}
-		}
+		l.labelSet[k] = struct{}{}
+	}
+}
+
+// AddFields saves field names so that they can be used to build the label fields in the exemplar frame.
+func (l *labelTracker) AddFields(fields []*data.Field) {
+	for _, f := range fields {
```

Comment:
Since `f` is a pointer do you think we should do a nil check? 

```
if f != nil {
	l.labelSet[f.Name] = struct{}{}
}
```

Reply:
I think it is okay to skip this. `[]*data.Field` and `[]*data.Frame` (the slices) - as best as I understand it now, was a mistake to make them pointers. So `nils` should not be inserted into those slices, and if they are, the datasource likely has a programming bug.

So not checking is more fragile, but they really never should be nil inside the slice.

Reply:
https://github.com/grafana/grafana-plugin-sdk-go/pull/618